tests/functional_tests/cpp_backend_tests/tridiagonal_solve_driver.cpp

```diff
@@ -0,0 +1,52 @@
+#include <gtest/gtest.h>
+
+#include GENERATED_FILE
+
+#include <fn_select.hpp>
+#include <gridtools/sid/rename_dimensions.hpp>
+#include <test_environment.hpp>
+
+namespace {
+using namespace gridtools;
+using namespace fn;
+using namespace literals;
+
+constexpr inline auto a = [](auto...) { return -1; };
+constexpr inline auto b = [](auto...) { return 3; };
+constexpr inline auto c = [](auto...) { return 1; };
+constexpr inline auto d = [](int ksize) {
+  return [kmax = ksize - 1](auto, auto, auto k) {
+    return k == 0 ? 4 : k == kmax ? 2 : 3;
+  };
+};
```

Comment:
reminder to back-port this change to gridtools c++ example

Reply:
It can’t be backported to GT C++, this simplification was only possible because we only test the Cartesian case here.